### ISSUE TITLE
Request for information

### ISSUE THREAD

**Issue by Srini-1981** [holms-ur/fine-tuning#15]
Hello Joheras,
Could not get much information on this transfer learning, need bit of clarification for below two lines of code.
- Train weights (output layers or 'heads')
model.train(train_set, test_set, learning_rate=config.LEARNING_RATE, epochs=2, layers='heads')
- Unfreeze the body of the network and train *all* layers
model.train(train_set, test_set, epochs=5, layers="all", learning_rate=config.LEARNING_RATE / 10)

Training heads is enough for table detection during fine tuning phase or we need to train all the layers? What is the difference between these two in model performance? From your initial training, what is the optimal number of epochs for a good table detection model convergence?

Might be too many questions, thanks again for the help.

**Comment by joheras** [comment]
Hi,

I think that both questions are related. I don't have magic numbers since hiperparameters like the number of epochs are highly dependent on the dataset that you are using. 

The approach that I usually follow is first train only the heads layers until the mAP for the validation set does not improve, and then start training the whole body. 

In some cases, it might be enough with training only the heads layers if the tables are quite similar to those appearing in the tablebank dataset; otherwise you might need to try unfreezing and training all the layers. 

Hope this helps.

**Comment by Srini-1981** [comment]
Thanks Joheras and it make sense. We just want to train only on our custom data without tablebank weights, is it possible  with Mask-RCNN? I see in many examples they are loading the coco weights (mask_rcnn_coco.h5) and train on the custom dataset, this is a mandatory step? What steps you took to train table bank weights, mean to ask you used any pretrained weights before training the model with tablebank data? Thanks.

**Comment by joheras** [comment]
Hi, If you have enough images, you can train your models from scratch without using the mask_rcnn_coco.h5 weights or the tablebank weights. 

This was the case for the tablebank dataset where we trained the model without loading any pretrained weights. In that case, we trained the whole network from the beginning (that is, we did not start training the heads layers and then the others). 

Hope this helps.

**Comment by Srini-1981** [comment]
Thanks Joheras and apologies for late reply, as of now fine tuning is going fine from our end. Will get back for any clarifications.
